examples/hooks/components/RangeInput.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from 'react';
+
+import { useRange, UseRangeProps } from 'react-instantsearch-hooks';
+import { cx } from '../cx';
+
+export type RangeProps = React.ComponentProps<'div'> & UseRangeProps;
+
+// if the default value is undefined, React considers the component uncontrolled initially, which we don't want 0 or NaN as the default value
+const controlledUndefined = '' as unknown as number;
+
+export function RangeInput(props: RangeProps) {
+  const {
+    range: { min, max },
+    start: [minValue, maxValue],
+    canRefine,
+    refine,
+  } = useRange(props);
+  const values = {
+    min:
+      minValue !== -Infinity && minValue !== min
+        ? minValue
+        : controlledUndefined,
+    max:
+      maxValue !== Infinity && maxValue !== max
+        ? maxValue
+        : controlledUndefined,
+  };
+  const [{ from, to }, setState] = useState({
```

Comment:
Naming suggestion, to narrow it down:

```suggestion
  const [{ from, to }, setRange] = useState({
```